### PR TITLE
Correct event validation messages using the core/v1 field name `reportingComponent`

### PR DESCRIPTION
**Pull request by sanwishe**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

when creating a events without `reportingComponent` field, the error message shows that `reportingController` is missing
```
The Event "xxx" is invalid:
* reportingController: Required value
* reportingController: Invalid value: "": name part must be non-empty
* reportingController: Invalid value: "": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yangjunmyfm192085 — COMMENTED** [review]
/ok-to-test

**Review by yangjunmyfm192085 — COMMENTED** [review]
/triage accepted

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Review by yangjunmyfm192085 — COMMENTED** [review]
/assign @liggitt

File: pkg/apis/core/validation/events.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func ValidateEventUpdate(newEvent, oldEvent *core.Event, requestVersion schema.G
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.EventTime, oldEvent.EventTime, field.NewPath("eventTime"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.Action, oldEvent.Action, field.NewPath("action"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.Related, oldEvent.Related, field.NewPath("related"))...)
-	allErrs = append(allErrs, ValidateImmutableField(newEvent.ReportingController, oldEvent.ReportingController, field.NewPath("reportingController"))...)
```
Comment (liggitt) [inline pkg/apis/core/validation/events.go:101]: this path only gets invoked for events.k8s.io requests (core/v1 requests return on line 79 above), so using the events/v1 field name here is correct

File: pkg/apis/core/validation/events.go [hunk 2]
```diff
@@ -144,9 +144,9 @@ func legacyValidateEvent(event *core.Event) field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("involvedObject", "namespace"), event.InvolvedObject.Namespace, "does not match event.namespace"))
 		}
 		if len(event.ReportingController) == 0 {
-			allErrs = append(allErrs, field.Required(field.NewPath("reportingController"), ""))
```
Comment (liggitt) [inline pkg/apis/core/validation/events.go:147]: the field name used should depend on the request version (which is plumbed to `ValidateEventUpdate` as `requestVersion`)... if this is via core/v1, use reportingComponent, if it's via events.k8s.io, use reportingController

Comment (sanwishe) [reply pkg/apis/core/validation/events.go:147]: Got it and already updated, would you please take a review please? thanks

**Pull request merged**